### PR TITLE
📐 Add C4 model diagrams

### LINKED ISSUE #580: 📐 Add C4 Diagrams to our Technical Documentation
## User Story

As a Data Platform team member
I want to view the platforms technical vision
So that I can understand our systems and explain them to users at differing technical 

## Definition of done

- [x] user docs have been updated
- [x] another team member has reviewed

## Reference

[How to write good user stories](https://www.gov.uk/service-manual/agile-delivery/writing-user-stories)

### PR DESCRIPTION
**Pull request by bagg3rs**
This Pull Request adds our c4 model diagrams to our technical documentation.

closes #580

### CODE REVIEW

File: docs/source/documentation/diagrams/architecture-diagrams.html.md.erb [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# Data Platform - Architectural Diagrams
```
Comment (hrahim-moj) [inline docs/source/documentation/diagrams/architecture-diagrams.html.md.erb]: No dash needed and sentence case for titles: Data Platform architectural diagrams

Comment (hrahim-moj) [reply docs/source/documentation/diagrams/architecture-diagrams.html.md.erb:1]: I'd include a sentence here that outlines the diagrams this page covers, as an overview.

File: docs/source/documentation/diagrams/architecture-diagrams.html.md.erb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+## C4 Model Diagrams
+
+In the [C4 model](https://c4model.com/), the four C's represent [context](https://c4model.com/#SystemContextDiagram), [containers](https://c4model.com/#ContainerDiagram), [components](https://c4model.com/#ComponentDiagram), and [code](https://c4model.com/#CodeDiagram). These different levels help you to tell the right story to the right audience.
```
Comment (hrahim-moj) [inline docs/source/documentation/diagrams/architecture-diagrams.html.md.erb]: I'd clarify the story and audience here.

'These different levels help you tell the right story to the right audience. In this case, for the Data Platform to its prospective users respectively.' [or to whoever the audience is if not them]

File: docs/source/documentation/diagrams/architecture-diagrams.html.md.erb [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+### Level 1 - Context Diagram
+This is the highest level of abstraction and provides an overview of the entire system. It shows the system under consideration and its external dependencies, such as users, external systems, and APIs. Frequency in change will be minimal compared to other levels.
```
Comment (hrahim-moj) [inline docs/source/documentation/diagrams/architecture-diagrams.html.md.erb]: Full-blown pedantry - just eradicating passive voice:

'Frequency in change is likely minimal compared to other levels.'

File: docs/source/documentation/diagrams/architecture-diagrams.html.md.erb (hunk 1 above)
Comment (hrahim-moj) [reply docs/source/documentation/diagrams/architecture-diagrams.html.md.erb]: Is it worth adding in a sentence here to introduce this pages as an index of architectural diagrams? I realise it only has the C4 ones so far, which are described below, but I'd include one if you add more diagrams to this page.

File: docs/source/documentation/diagrams/architecture-diagrams.html.md.erb [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+![Level 1 Diagram](c4-model-level1.excalidraw.png)
+
+### Level 2 - Container Diagram
+
+This level focuses on the software system's high-level [containers](https://c4model.com/#ContainerDiagram), which are typically applications or services. It illustrates the major components of the system and their interactions. Containers can be web servers, databases, desktop applications, mobile apps, etc.
```
Comment (hrahim-moj) [inline docs/source/documentation/diagrams/architecture-diagrams.html.md.erb]: Switching around to make active again:

Web servers, databases, desktop and mobile applications are all examples of containers.

File: docs/source/documentation/diagrams/architecture-diagrams.html.md.erb [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+####  3.1 Data Product Component - Proposal
+
+This diagram shows the relationships of a Data Product with other components of the Data Platform.
+
+![Level 3 Diagram](c4-model-level3-data-product.excalidraw.png)
+
+## Editing the diagrams
+
+These diagrams were created using the open source tool [Excalidraw](https://docs.excalidraw.com/). The source of the excalidraw files are embedded into the `.png` format creating a `.excalidraw.png` file.
```
Comment (hrahim-moj) [inline docs/source/documentation/diagrams/architecture-diagrams.html.md.erb]: We created these diagrams using the open-source tool Excalidraw. We then embedded the diagram source files into the .png format, which creates a .excalidraw.png file.

File: docs/source/documentation/diagrams/architecture-diagrams.html.md.erb [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+
+- On the web using [vscode.dev](https://github.dev/ministryofjustice/data-platform/tree/main/docs/diagrams) with the [Excalidraw extenstion](https://marketplace.visualstudio.com/items?itemName=pomdtr.excalidraw-editor)
```
Comment (hrahim-moj) [inline docs/source/documentation/diagrams/architecture-diagrams.html.md.erb:36]: It looks like this bullet point and the next one need an intro. Maybe just something as simple as 'There are multiple ways you can use Excalidraw:'

**Review by jacobwoffenden — APPROVED** [review]

**Pull request merged**